go/adapter/datadog/adapter.go

```diff
@@ -25,57 +28,68 @@ func DefaultDatadogConfig() *DatadogConfig {
 		ServiceName: "my-wasm-service",
 		DefaultTags: nil,
 		TraceType:   datadog_formatter.Web,
+		FlushPeriod: 2 * time.Second,
+		BatchSize:   100,
 	}
 }
 
 type DatadogAdapter struct {
-	observe.AdapterBase
+	*observe.AdapterBase
 	Config *DatadogConfig
 }
 
-func NewDatadogAdapter(config *DatadogConfig) (DatadogAdapter, error) {
+func NewDatadogAdapter(config *DatadogConfig) (*DatadogAdapter, error) {
 	if config == nil {
 		config = DefaultDatadogConfig()
 	}
 
-	return DatadogAdapter{
-		AdapterBase: observe.NewAdapterBase(),
+	base := observe.NewAdapterBase(config.BatchSize, config.FlushPeriod)
```

Comment:
unless there is a reason to use `int` for FlushPeriod in the AdapterBase, would definitely go with `time.Duration` -- haven't yet got to where this is used though, so apologies if it will become apparent :laughing: